TiUP: add instructions for offline mirror (#6137)
pingcap/docs: production-deployment-using-tiup.md

#### Step 1: Prepare the TiUP offline component package

> Option 1: Select the corresponding version of the package of the TiDB server's offline mirrors (including the TiUP offline component package) on the [Download TiDB page](https://pingcap.com/download/).

Author: Although the former version does not have this part, I added it temporarily because the updated part that needed translation mentioned it. If we don't need this one, should we also revise the translation "If the package of the TiDB server's offline mirrors from the PingCAP website does not meet your specific needs," in the below part?

Reviewer: This part is not needed in the English version. You cannot "Select the corresponding version of the package" in https://pingcap.com/download/. The English version only provides the `tiup mirror clone` download method. Please translate the docs-cn PR **AS IT IS**.

        `tidb-community-server-${version}-linux-amd64.tar.gz` is an independent offline environment package.

> 3. Customize the offline mirror, or adjust the contents of the existing offline  mirror.

Reviewer 2: ```suggestion
3. Customize the offline mirror, or adjust the contents of an existing offline  mirror.
```

>     If you want to adjust the existing offline mirror (such as adding a new version of a component), take the following steps:

Reviewer 2: ```suggestion
    If you want to adjust an existing offline mirror (such as adding a new version of a component), take the following steps:
```

>     If the existing offline mirrors does not meet your specific needs, or if you want to make adjustments to the mirror (like adding a new version of a component), take the following process:

Author: I have revised the first line of this part because we don't provide the package that can be downloaded from our website, please take a look.

Reviewer: ```suggestion
    If you want to adjust the existing offline mirror (such as adding a new version of a component), take the following steps:
```

    1. When pulling an offline mirror, you can get an incomplete offline mirror by specifying specific information via parameters, such as the component and version information. For example, you can pull an offline mirror that includes only the offline mirror of TiUP v1.5.2 and TiUP Cluster v1.5.2 by running the following command:

        {{< copyable "shell-regular" >}}

        ```bash
        tiup mirror clone tiup-custom-mirror-v1.5.2 --tiup v1.5.2 --cluster v1.5.2
        ```

>         If you only need the components for a particular platform, you can specify them using the `--os`  or `--arch` parameter.

Reviewer 2: ```suggestion
        If you only need the components for a particular platform, you can specify them using the `--os`  or `--arch` parameters.
```

>     2. Refer to the step 2 of "Pull the mirror using TiUP", and send this incomplete offline mirror to the control machine in the isolated environment.

Reviewer 2: ```suggestion
    2. Refer to the step 2 of "Pull the mirror using TiUP", and send this incomplete offline mirror to the controller machine in the isolated environment.
```

or i think "console machine" seems ok too

Author: we used to translate "中控机“ into "control machine" (for example, in the document [Upgrade TiDB Using TiUP](https://docs.pingcap.com/tidb/stable/upgrade-tidb-using-tiup) and [tiup cluster list](https://docs.pingcap.com/tidb/stable/tiup-component-cluster-list#tiup-cluster-list). and "controller" used as the translation of " "控制工具" (in [Deploy and Maintain an Online TiDB Cluster Using TiUP](https://docs.pingcap.com/zh/tidb/stable/tiup-cluster). Could we keep it as "the control machine" for the consistency?

Author: @AstroProfundis

Reviewer 2: ok then, I didn't notice about that before.

>     2. Refer to step 2 mentioned above, "Install TiUP on the control machine",  send this incomplete offline mirror to the control machine in the isolated environment.

Reviewer: ```suggestion
    2. Refer to the step 2 of "Pull the mirror using TiUP", and send this incomplete offline mirror to the control machine in the isolated environment.
```

>     3. Check the path of the current offline mirror on the control machine in the isolated environment. If your TiUP tool is of a newer version, you can get the current mirror address by running the following command:

Reviewer 2: ```suggestion
    3. Check the path of the current offline mirror on the control machine in the isolated environment. If your TiUP tool is of a recent version, you can get the current mirror address by running the following command:
```

>     3. Check the path of the current offline mirror on the control machine in the isolated environment. If you installed TiUP in the newer versions, you can get the current mirror address by running the following command:

Reviewer: ```suggestion
    3. Check the path of the current offline mirror on the control machine in the isolated environment. If your TiUP tool is of a newer version, you can get the current mirror address by running the following command:
```

        {{< copyable "shell-regular" >}}

        ```bash
        tiup mirror show
        ```

>         If the above command indicates that the `show` command does not exist, you may now use an older version of TiUP. In this case, you can get the current mirror address from `$HOME/.tiup/tiup.toml`. Write down this mirror address, and then use `${base_mirror}` to refer to it.

Reviewer: ```suggestion
        If the output of the above command indicates that the `show` command does not exist, you might be using an older version of TiUP. In this case, you can get the current mirror address from `$HOME/.tiup/tiup.toml`. Record this mirror address. In the following steps, `${base_mirror}` is used to refer to this address.
```

    4. Merge an incomplete offline mirror into an existing offline mirror:

        First, copy the `keys` directory in the current offline mirror to the `$HOME/.tiup` directory:

        {{< copyable "shell-regular" >}}

        ```bash
        cp -r ${base_mirror}/keys $HOME/.tiup/
        ```

>         Then use the TiUP command to merge the incomplete offline mirror into the currently used mirror:

Reviewer: ```suggestion
        Then use the TiUP command to merge the incomplete offline mirror into the mirror in use:
```

        {{< copyable "shell-regular" >}}

        ```bash
        tiup mirror merge tiup-custom-mirror-v1.5.2
        ```
    

>     5. When the above steps are completed, check the result by running the `tiup list` command. In this document's example, the outputs of both `tiup list tiup` and `tiup list cluster` show that the version of the corresponding component is `v1.5.2`.

Reviewer 2: ```suggestion
    5. When the above steps are completed, check the result by running the `tiup list` command. In this document's example, the outputs of both `tiup list tiup` and `tiup list cluster` show that `v1.5.2` of the corresponding components are available.
```

Author: how about "... that the corresponding components in `v1.5.2` are available."? @AstroProfundis @TomShawn

Reviewer 2: lgtm

Reviewer: lgtm

Reviewer: corresponding components in v1.5.2  --> corresponding components of v1.5.2

>     5. When the above steps are complete, check the results by running the `tiup list`command. In this document's example, you can find the `v1.5.2` version of the corresponding components in the outputs of `tiup list tiup` and `tiup list cluster`.

Reviewer: ```suggestion
    5. When the above steps are completed, check the result by running the `tiup list` command. In this document's example, the outputs of both `tiup list tiup` and `tiup list cluster` show that the version of the corresponding component is `v1.5.2`.
```
